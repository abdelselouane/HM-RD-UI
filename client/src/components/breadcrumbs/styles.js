import styled from 'styled-components';

export const Container = styled.div`
  grid-area: cmess;

  .breadcrumb {
    font-size: 16px;
  }
`;

export default Container;
