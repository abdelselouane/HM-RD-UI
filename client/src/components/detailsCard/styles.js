import styled from 'styled-components';

export const Card = styled.div`
  width: 85%;
  margin-bottom: 20px;

  .card-content {
    padding: 0px 25px;
  }
`;

export const Title = styled.span`
  align-items: flex-end;
  display: flex;
  justify-content: flex-start;
  width: 100%;
  border-bottom: solid #666 1px;
  padding-bottom: 15px;

  h2 {
    display: inline-block;
    color: #f96302;
    font-weight: 900;
    padding-right: 25px;
  }

  span {
    text-transform: capitalize;
  }
`;

export const Row = styled.div`
  display: flex;
  padding: 15px;
`;

export const Col = styled.div`
  ${props => props.flex && `flex: ${props.flex}`}
  display: flex;
  flex-direction: column;
`;

export const Label = styled.div`
  font-weight: 700;
  color: #666;
`;
