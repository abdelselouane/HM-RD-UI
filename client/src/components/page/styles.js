import styled, { css } from 'styled-components';

const defaultGrid = css`
  display: grid;
  grid-template-rows: 70px
                      48px
                      24px
                      24px
                      repeat(3, 1fr)
                      repeat(1, 72px);
  grid-template-columns: 20px 1fr repeat(4, 3fr) 1fr 20px;
  grid-template-areas:  ". . . . . . . ."
                        ". . . . . . . ."
                        ". . cmess cmess cmess cmess . ."
                        ". . cmess cmess cmess cmess . ."
                        ". . carea carea carea carea . ."
                        ". . carea carea carea carea . ."
                        ". . carea carea carea carea . ."
                        ". . . . . . . .";
`;

const lowResolutionGrid = css`
  display: grid;
  grid-template-areas:  ". . . . . . . ."
                        ". . . . . . . ."
                        ". cmess cmess cmess cmess cmess cmess ."
                        ". cmess cmess cmess cmess cmess cmess ."
                        ". carea carea carea carea carea carea ."
                        ". carea carea carea carea carea carea ."
                        ". carea carea carea carea carea carea ."
                        ". . . . . . . .";
  grid-template-rows: 70px
                      24px
                      24px
                      24px
                      repeat(3, 1fr)
                      repeat(1, 48px);
  grid-template-columns: 10px 1fr repeat(4, 3fr) 1fr 10px;
`;

export const GridLayout = styled.div`
  background-color: ${({ theme }) => theme.black12};
  box-sizing: border-box;
  color: ${({ theme }) => theme.blackA11};
  height: 100%;  
  ${props => (props.enableScroll ? 'overflow-x: scroll;' : 'overflow: hidden;')}
  
  ${() => defaultGrid}
  @media only screen and (max-width: 1024px) {
    ${() => lowResolutionGrid}
  }
`;

export const Container = styled.div`
  align-items: flex-start;
  display: flex;
  grid-area: carea;
  justify-content: center;
  ${({ hideBoxShadow }) => (hideBoxShadow ? '' : 'box-shadow: 0 0 2px 0 rgba(0, 0, 0, 0.12), 0 2px 2px 0 rgba(0, 0, 0, 0.24);')}
`;
