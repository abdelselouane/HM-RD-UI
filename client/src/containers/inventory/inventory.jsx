import React from 'react';
import kebabcase from 'lodash.kebabcase';
import styled from 'styled-components';

import Container from './styles';

import Tabs from '../../components/tabs';
import Page from '../../components/page';

import ReceiveOrders from '../receiveOrders';
import ManageInventory from '../manageInventory';

export const Content = styled.div`
  padding: 20px;
`;

export const Inventory = () => {
  const INVENTORY_TABS = ['On Hand Inventory', 'Receive Purchase Orders'];
  const tabs = [
    {
      displayName: INVENTORY_TABS[0],
      id: kebabcase(INVENTORY_TABS[0]),
      component: ManageInventory,
    },
    {
      displayName: INVENTORY_TABS[1],
      id: kebabcase(INVENTORY_TABS[1]),
      component: ReceiveOrders,
    }
  ];

  return (
    <Page className="inventory">
      <Container>
        <Tabs
          className="inventory-tabs"
          tabs={tabs}
          defaultTab={tabs[0].id}
        />
      </Container>
    </Page>
  );
};

export default Inventory;
