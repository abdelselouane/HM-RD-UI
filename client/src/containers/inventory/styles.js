import { Container as BaseContainer } from '../../components/page';

export const Container = BaseContainer.extend`
background-color: ${({ theme }) => theme.white};
box-sizing: border-box;
display: flex;
flex-direction: column;
justify-content: flex-start;
height: 100%;
overflow-y: hidden;
width: 100%;


.inventory-tabs {
  height: 100%;
}

.tabs span {
  &:focus {
    outline: none !important;
  }
  
  &[role=tab]:first-child {
    border-left: none;
  }
}

[role=tabpanel] {
    display: flex;
    flex-direction: column;
    height: 95%;
    overflow: hidden;
    padding: 20px 0px;

    .text-input-container.search {
      margin: 15px 15px;
      width: 65%;

      &.dropdown {
        width: 75%;
        margin: 15px 0px;
      }
    }
}

.search-container{
  display: flex;
  flex-direction: row;
  justify-content: space-between;

  button {
    flex: 0.15;
  }
}

.text-input-container .input-field {
  margin-top: 0em;
}

.edit-col .button {
  margin: 0px;
}

.text-input-container.search {
  flex: 0.5;

  input {
    margin-top: -6px;
  }

  .icon_search {
    bottom: 0;
    color: ${({ theme }) => theme.orange};
    font-size: 30px;
    padding: 5px 10px;
    position: absolute;
    right: 0;
    z-index: 1;
  }
}
`;

export default Container;
