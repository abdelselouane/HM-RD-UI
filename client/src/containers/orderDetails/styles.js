import styled from 'styled-components';
import { Container as BaseContainer } from '../../components/page';

export const Container = BaseContainer.extend`
  background-color: ${({ theme }) => theme.white};
  align-items: flex-start;
  display: flex;
  flex-direction: column;
`;

export const CardTitle = styled.div`
  align-items: flex-end;
  display: flex;
  justify-content: space-between;
  height: 50px;
  margin: 0px;
  margin-bottom: 30px;
  padding: 0px 20px;
  width: 100%;

  h2 {
    display: inline-block;
    margin-bottom: 0px;
  }
`;

export const PurchaseOrderList = styled.div`
  align-items: center;
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  padding-top: 20px;
  flex: 1;
  overflow-y: scroll;
  overflow-x: hidden;
  width: 100%;
`;
