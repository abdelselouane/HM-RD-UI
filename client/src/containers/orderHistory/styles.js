import { Container as BaseContainer } from '../../components/page';

export const Container = BaseContainer.extend`
  background-color: ${({ theme }) => theme.white};
  align-items: flex-start;
  display: flex;
  flex-direction: column;

  .title {
    padding: 0px 20px;
  }
  
  .order-history-header {
    margin: 0px;
  
    thead tr {
      margin-left: 1em;
  
      th {
        border: none;
        font-size: 18px;
  
        &:first-child {
          padding-left: 20px;
        }
      }
    }
  }
`;

export default Container;
