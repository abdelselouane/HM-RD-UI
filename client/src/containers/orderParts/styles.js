import { Container as BaseContainer } from '../../components/page';

export const Container = BaseContainer.extend`
background: ${({ theme }) => theme.white};
`;

export default Container;
