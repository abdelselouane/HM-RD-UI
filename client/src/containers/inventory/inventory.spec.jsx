/* global describe, beforeAll, it, expect */
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { Inventory } from './inventory';
import Tabs from '../../components/tabs';
import * as Styles from './styles';

describe('<Inventory />', () => {
  let wrapper;
  let props;

  beforeAll(() => {
    wrapper = shallow(<Inventory {...props} />);
  });

  it('should render properly', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render styles properly', () => {
    Object.values(Styles).forEach(Component => {
      expect(toJson(shallow(<Component />))).toMatchSnapshot();
    });
  });

  it('should render receive orders tab', () => {
    expect(wrapper.find(Tabs).props().tabs[1].component.displayName).toBe('Connect(ReceiveOrders)');
  });
});
