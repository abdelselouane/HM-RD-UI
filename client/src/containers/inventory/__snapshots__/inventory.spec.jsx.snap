// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Inventory /> should render properly 1`] = `
<Connect(withContext(Page))
  className="inventory"
>
  <styled.div>
    <Tabs
      className="inventory-tabs"
      defaultTab="on-hand-inventory"
      tabs={
        Array [
          Object {
            "component": [Function],
            "displayName": "On Hand Inventory",
            "id": "on-hand-inventory",
          },
          Object {
            "component": [Function],
            "displayName": "Receive Purchase Orders",
            "id": "receive-purchase-orders",
          },
        ]
      }
    />
  </styled.div>
</Connect(withContext(Page))>
`;

exports[`<Inventory /> should render styles properly 1`] = `
.c0 {
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  grid-area: carea;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  box-shadow: 0 0 2px 0 rgba(0,0,0,0.12),0 2px 2px 0 rgba(0,0,0,0.24);
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  height: 100%;
  overflow-y: hidden;
  width: 100%;
}

.c0 .inventory-tabs {
  height: 100%;
}

.c0 .tabs span:focus {
  outline: none !important;
}

.c0 .tabs span[role=tab]:first-child {
  border-left: none;
}

.c0 [role=tabpanel] {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 95%;
  overflow: hidden;
  padding: 20px 0px;
}

.c0 [role=tabpanel] .text-input-container.search {
  margin: 15px 15px;
  width: 65%;
}

.c0 [role=tabpanel] .text-input-container.search.dropdown {
  width: 75%;
  margin: 15px 0px;
}

.c0 .search-container {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c0 .search-container button {
  -webkit-flex: 0.15;
  -ms-flex: 0.15;
  flex: 0.15;
}

.c0 .text-input-container .input-field {
  margin-top: 0em;
}

.c0 .edit-col .button {
  margin: 0px;
}

.c0 .text-input-container.search {
  -webkit-flex: 0.5;
  -ms-flex: 0.5;
  flex: 0.5;
}

.c0 .text-input-container.search input {
  margin-top: -6px;
}

.c0 .text-input-container.search .icon_search {
  bottom: 0;
  font-size: 30px;
  padding: 5px 10px;
  position: absolute;
  right: 0;
  z-index: 1;
}

<div
  className="c0"
/>
`;

exports[`<Inventory /> should render styles properly 2`] = `
.c0 {
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  grid-area: carea;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  box-shadow: 0 0 2px 0 rgba(0,0,0,0.12),0 2px 2px 0 rgba(0,0,0,0.24);
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  height: 100%;
  overflow-y: hidden;
  width: 100%;
}

.c0 .inventory-tabs {
  height: 100%;
}

.c0 .tabs span:focus {
  outline: none !important;
}

.c0 .tabs span[role=tab]:first-child {
  border-left: none;
}

.c0 [role=tabpanel] {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 95%;
  overflow: hidden;
  padding: 20px 0px;
}

.c0 [role=tabpanel] .text-input-container.search {
  margin: 15px 15px;
  width: 65%;
}

.c0 [role=tabpanel] .text-input-container.search.dropdown {
  width: 75%;
  margin: 15px 0px;
}

.c0 .search-container {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c0 .search-container button {
  -webkit-flex: 0.15;
  -ms-flex: 0.15;
  flex: 0.15;
}

.c0 .text-input-container .input-field {
  margin-top: 0em;
}

.c0 .edit-col .button {
  margin: 0px;
}

.c0 .text-input-container.search {
  -webkit-flex: 0.5;
  -ms-flex: 0.5;
  flex: 0.5;
}

.c0 .text-input-container.search input {
  margin-top: -6px;
}

.c0 .text-input-container.search .icon_search {
  bottom: 0;
  font-size: 30px;
  padding: 5px 10px;
  position: absolute;
  right: 0;
  z-index: 1;
}

<div
  className="c0"
/>
`;
